src/App.tsx

```diff
@@ -1,24 +1,234 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  useCallback, useEffect, useState, FC, useMemo,
+} from 'react';
+import cn from 'classnames';
+import {
+  deleteTodo, getTodos, patchTodo, postTodo,
+} from './api/todos';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem/TodoItem';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+
+import { TodoData } from './types/TodoData';
+import { ErrorBy } from './types/ErrorBy';
+import { FilterBy } from './types/FilterBy';
+
+const USER_ID = 10331;
+
+export const App: FC = () => {
+  const [todoTemp, setTodoTemp] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [typeOfFilter, setTypeOfFilter] = useState<FilterBy>(FilterBy.all);
+  const [disableWriting, setDisableWriting] = useState(false);
+  const [typeOfError, setTypeOfError] = useState<ErrorBy | null>(null);
+  const [isError, setIsError] = useState(false);
+  const [isClearCompletedTodos, setIsClearCompletedTodos] = useState(false);
+
+  const completedTodos = useMemo(() => {
+    return todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+  }, [todos]);
+
+  const numberOfItemsLeft = useMemo(() => (
+    todos.filter(({ completed }) => !completed).length
+  ), [todos]);
+
+  const hasCompletedTodos = useMemo(() => (
+    numberOfItemsLeft !== todos.length
+  ), [todos]);
+
+  const hasTodos = useMemo(() => todos.length > 0, [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter(({ completed }) => {
+      switch (typeOfFilter) {
+        case FilterBy.active:
+          return !completed;
+        case FilterBy.completed:
+          return completed;
+        default:
+          return true;
+      }
+    });
+  }, [todos, typeOfFilter]);
+
+  const clearTitle = useCallback(() => setTitle(''), []);
+
+  const loadTodos = useCallback(async () => {
+    setIsError(false);
+    setDisableWriting(true);
+
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setIsError(true);
+      setTypeOfError(ErrorBy.loading);
+    }
+
+    setDisableWriting(false);
+  }, []);
+
+  const handleAddTodo = useCallback(async (
+    event: React.ChangeEvent<HTMLFormElement>,
+  ) => {
+    event.preventDefault();
+
+    const newTodo: TodoData = {
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    try {
+      if (typeOfError === ErrorBy.loading || typeOfError === ErrorBy.adding) {
+        setTypeOfError(ErrorBy.adding);
+        throw new Error();
+      }
+
+      if (!title.trim()) {
+        setTypeOfError(ErrorBy.todoIsEmpty);
+        throw new Error();
+      }
 
-const USER_ID = 0;
+      const temporaryTodo = { ...newTodo, id: 0 };
```

Comment:
maybe instead of 0, try to use null
who knows, maybe server also give you an id equal to 0